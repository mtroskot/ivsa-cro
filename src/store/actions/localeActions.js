import { localeActionTypes } from 'src/constants/actionTypes';

export function changeLocale(locale) {
  return {
    type: localeActionTypes.CHANGE_LOCALE,
    payload: { locale }
  };
}
export function changeLocaleSuccess(locale) {
  return {
    type: localeActionTypes.CHANGE_LOCALE_SUCCESS,
    payload: { locale }
  };
}
