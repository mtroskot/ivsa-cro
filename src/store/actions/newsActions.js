import { newsActionTypes } from 'src/constants/actionTypes';

export function fetchNews(refreshing = false) {
  return {
    type: newsActionTypes.FETCH_NEWS,
    payload: { refreshing }
  };
}
export function fetchNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.FETCH_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}
export function fetchNewsError(error) {
  return {
    type: newsActionTypes.FETCH_NEWS_ERROR,
    payload: { error }
  };
}

export function publishNews(date, publishedBy, notice) {
  return {
    type: newsActionTypes.PUBLISH_NEWS,
    payload: { date, publishedBy, notice }
  };
}

export function publishNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.PUBLISH_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}

export function updateNews(newsId, date, publishedBy, notice) {
  return {
    type: newsActionTypes.UPDATE_NEWS,
    payload: { newsId, date, publishedBy, notice }
  };
}

export function updateNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.UPDATE_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}

export function deleteNews(newsId) {
  return {
    type: newsActionTypes.DELETE_NEWS,
    payload: { newsId }
  };
}

export function deleteNewsSuccess(news, lastUpdate) {
  return {
    type: newsActionTypes.DELETE_NEWS_SUCCESS,
    payload: { news, lastUpdate }
  };
}
