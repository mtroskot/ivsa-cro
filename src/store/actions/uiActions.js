import { uiActionTypes } from 'src/constants/actionTypes';
import { ObjectUtils } from 'src/utils';

export const startAction = (name, params) => {
  if (!ObjectUtils.isObject(params)) {
    throw Error(`type of ${typeof params} is not an object`);
  }
  return {
    type: uiActionTypes.START_ACTION,
    payload: {
      action: {
        name,
        params
      }
    }
  };
};

export const stopAction = name => ({
  type: uiActionTypes.STOP_ACTION,
  payload: { name }
});

export const refreshActionStart = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_START,
  payload: { refreshAction }
});

export const refreshActionStop = refreshAction => ({
  type: uiActionTypes.REFRESH_ACTION_STOP,
  payload: { refreshAction }
});

export const togglePopupMessage = (message, position) => ({
  type: uiActionTypes.TOGGLE_POPUP_MESSAGE,
  payload: {
    message,
    position
  }
});
