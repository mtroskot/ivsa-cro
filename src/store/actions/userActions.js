import { userActionTypes } from 'src/constants/actionTypes';

export function authenticateUser(email, password) {
  return {
    type: userActionTypes.AUTH_USER,
    payload: { email, password }
  };
}

export const authenticateUserSuccess = (displayName, email, idToken, expiryDate) => {
  return {
    type: userActionTypes.AUTH_USER_SUCCESS,
    payload: { displayName, email, idToken, expiryDate }
  };
};

export const authenticateUserError = error => {
  return {
    type: userActionTypes.AUTH_USER_ERROR,
    payload: { error }
  };
};

export const logout = () => {
  return {
    type: userActionTypes.LOG_OUT,
    payload: {}
  };
};
