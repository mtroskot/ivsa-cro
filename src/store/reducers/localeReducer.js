import { localeActionTypes } from 'src/constants/actionTypes';

export const initialState = {
  currLocale: 'en'
};

const localeReducer = (state = initialState, { type, payload }) => {
  switch (type) {
    case localeActionTypes.CHANGE_LOCALE_SUCCESS:
      return {
        ...state,
        currLocale: payload.locale
      };
    default:
      return state;
  }
};

export default localeReducer;
