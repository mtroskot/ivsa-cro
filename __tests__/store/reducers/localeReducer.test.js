import localeReducer, { initialState } from 'src/store/reducers/localeReducer';
import * as localeActions from 'src/store/actions/localeActions';

describe('INITIAL_STATE', () => {
  const beforeActionState = initialState;

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(localeReducer(beforeActionState, action)).toEqual(expectedState);
    expect(localeReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns changeLocaleSuccess state', () => {
    const locale = 'locale';
    const startAction = localeActions.changeLocaleSuccess(locale);
    const expectedState = {
      ...beforeActionState,
      currLocale: locale
    };
    expect(localeReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(localeReducer(beforeActionState, startAction)).toMatchSnapshot();
  });
});

describe('CHANGE_LOCALE_SUCCESS_STATE', () => {
  const beforeActionState = initialState;

  it('1returns initial state', () => {
    const action = { type: 'dummy_action' };
    const expectedState = beforeActionState;
    expect(localeReducer(beforeActionState, action)).toEqual(expectedState);
    expect(localeReducer(beforeActionState, action)).toMatchSnapshot();
  });

  it('2returns changeLocaleSuccess state', () => {
    const locale = 'locale2';
    const startAction = localeActions.changeLocaleSuccess(locale);
    const expectedState = {
      ...beforeActionState,
      currLocale: locale
    };
    expect(localeReducer(beforeActionState, startAction)).toEqual(expectedState);
    expect(localeReducer(beforeActionState, startAction)).toMatchSnapshot();
  });
});
