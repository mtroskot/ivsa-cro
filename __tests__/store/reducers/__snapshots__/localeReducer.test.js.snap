// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CHANGE_LOCALE_SUCCESS_STATE 1returns initial state 1`] = `
Object {
  "currLocale": "en",
}
`;

exports[`CHANGE_LOCALE_SUCCESS_STATE 2returns changeLocaleSuccess state 1`] = `
Object {
  "currLocale": "locale2",
}
`;

exports[`INITIAL_STATE 1returns initial state 1`] = `
Object {
  "currLocale": "en",
}
`;

exports[`INITIAL_STATE 2returns changeLocaleSuccess state 1`] = `
Object {
  "currLocale": "locale",
}
`;
